includes/Core/Permissions/Permissions.php

```diff
@@ -93,32 +114,51 @@ final class Permissions {
 	 *
 	 * @param Context        $context        Plugin context.
 	 * @param Authentication $authentication Optional. Authentication instance. Default is a new instance.
+	 * @param Modules        $modules        Optional. Modules instance. Default is a new instance.
 	 */
-	public function __construct( Context $context, Authentication $authentication = null ) {
+	public function __construct( Context $context, Authentication $authentication = null, Modules $modules = null ) {
```

Comment:
@jimmymadon, please, rework your changes to avoid instantiating new instances of `Modules`, `Module_Sharing_Settings`, `Authentication`, and other classes. All dependencies must be provided via constructor. To get the sharing settings instance, add a new public method to the `Modules` class that will return one defined and registered in that class.

Reply:
@eugene-manuilov Just wanted to check if should make the existing dependancy (Authentication) and new dependencies (Modules, User_Options) optional or mandatory?

Reply:
@jimmymadon Yes, let's make them mandatory. I think that the authentication instant is optional only because we don't want to bother with it in our phpunit tests. But in our codebase we always pass the authentication object when we create a new instance of the Permissions class.

Also, could you please update the `Debug_Data` class to receive an instance of the `Permissions` class via constructor as well instead of creating it internally?